From 10317939d461347e1383750f53b4cfeaf8da42dc Mon Sep 17 00:00:00 2001
From: Phil Smart <philip.smart@jisc.ac.uk>
Date: Thu, 22 Jul 2021 16:01:30 +0100
Subject: [PATCH] JPAR-178 - Update maven-javadoc-plugin to 3.3.0

Rebase maven-javadoc-plugin changes onto 3.3.0.

- Prevent the generation of module path arguments to Javadoc. Instead
always using the classpath argument.
- Change POM for internal project use
- Replace full javadoc goal with shib plugin groupId

https://issues.shibboleth.net/jira/browse/JPAR-178
---
 pom.xml                                       | 28 +++++++------------
 .../plugins/javadoc/AbstractJavadocMojo.java  | 18 +++++-------
 2 files changed, 17 insertions(+), 29 deletions(-)

diff --git a/pom.xml b/pom.xml
index 7e9ab315..c0baffdf 100644
--- a/pom.xml
+++ b/pom.xml
@@ -29,12 +29,14 @@ under the License.
     <relativePath />
   </parent>
 
+  <groupId>net.shibboleth.maven.plugins</groupId>
   <artifactId>maven-javadoc-plugin</artifactId>
   <version>3.3.0</version>
   <packaging>maven-plugin</packaging>
 
   <name>Apache Maven Javadoc Plugin</name>
   <description>
+    Patched version of The Apache Maven Javadoc Plugin, see JPAR-178.
     The Apache Maven Javadoc Plugin is a plugin that uses the javadoc tool for
     generating javadocs for the specified project.
   </description>
@@ -44,25 +46,15 @@ under the License.
     <maven>${mavenVersion}</maven>
   </prerequisites>
 
-  <scm>
-    <connection>scm:git:https://gitbox.apache.org/repos/asf/maven-javadoc-plugin.git</connection>
-    <developerConnection>scm:git:https://gitbox.apache.org/repos/asf/maven-javadoc-plugin.git</developerConnection>
-    <url>https://github.com/apache/maven-javadoc-plugin/tree/${project.scm.tag}</url>
-    <tag>maven-javadoc-plugin-3.3.0</tag>
-  </scm>
-  <issueManagement>
-    <system>jira</system>
-    <url>https://issues.apache.org/jira/browse/MJAVADOC</url>
-  </issueManagement>
-  <ciManagement>
-    <system>Jenkins</system>
-    <url>https://ci-builds.apache.org/job/Maven/job/maven-box/job/maven-javadoc-plugin/</url>
-  </ciManagement>
   <distributionManagement>
-    <site>
-      <id>apache.website</id>
-      <url>scm:svn:https://svn.apache.org/repos/asf/maven/website/components/${maven.site.path}</url>
-    </site>
+     <repository>
+         <id>release</id>
+         <url>https://build.shibboleth.net/nexus/content/repositories/releases</url>
+     </repository>
+     <snapshotRepository>
+         <id>snapshot</id>
+         <url>https://build.shibboleth.net/nexus/content/repositories/snapshots</url>
+     </snapshotRepository>
   </distributionManagement>
 
   <properties>
diff --git a/src/main/java/org/apache/maven/plugins/javadoc/AbstractJavadocMojo.java b/src/main/java/org/apache/maven/plugins/javadoc/AbstractJavadocMojo.java
index 0655f8a1..3af0eefb 100644
--- a/src/main/java/org/apache/maven/plugins/javadoc/AbstractJavadocMojo.java
+++ b/src/main/java/org/apache/maven/plugins/javadoc/AbstractJavadocMojo.java
@@ -5121,15 +5121,7 @@ public abstract class AbstractJavadocMojo
         Map<String, JavaModuleDescriptor> allModuleDescriptors = new HashMap<>();
 
         
-        boolean supportModulePath = javadocRuntimeVersion.isAtLeast( "9" );
-        if ( release != null ) 
-        {
-            supportModulePath &= JavaVersion.parse( release ).isAtLeast( "9" );
-        }
-        else if ( source != null )
-        {
-            supportModulePath &= JavaVersion.parse( source ).isAtLeast( "9" );
-        }
+        boolean supportModulePath = false;
 
         if ( supportModulePath )
         {
@@ -5392,6 +5384,7 @@ public abstract class AbstractJavadocMojo
         }
         else
         {
+            getLog().info( " Building classpath javadoc argument" );
             String classpath = StringUtils.join( getPathElements().iterator(), File.pathSeparator );
             addArgIfNotEmpty( arguments, "-classpath", JavadocUtil.quotedPathArgument( classpath ) , false, false );
         }
@@ -5989,6 +5982,7 @@ public abstract class AbstractJavadocMojo
     private void doExecuteJavadocCommandLine( Commandline cmd, File javadocOutputDirectory )
         throws MavenReportException
     {
+        getLog().info( "Running patched (for flat multi-module projects) maven javadoc plugin" );
         if ( getLog().isDebugEnabled() )
         {
             // no quoted arguments
@@ -6281,7 +6275,7 @@ public abstract class AbstractJavadocMojo
     private String getFullJavadocGoal()
     {
         String javadocPluginVersion = null;
-        String resource = "META-INF/maven/org.apache.maven.plugins/maven-javadoc-plugin/pom.properties";
+        String resource = "META-INF/maven/net.shibboleth.maven.plugins/maven-javadoc-plugin/pom.properties";
         try ( InputStream resourceAsStream 
                         = AbstractJavadocMojo.class.getClassLoader().getResourceAsStream( resource ) )
         {
@@ -6302,7 +6296,7 @@ public abstract class AbstractJavadocMojo
 
         StringBuilder sb = new StringBuilder();
 
-        sb.append( "org.apache.maven.plugins:maven-javadoc-plugin:" );
+        sb.append( "net.shibboleth.maven.plugins:maven-javadoc-plugin:" );
         if ( StringUtils.isNotEmpty( javadocPluginVersion ) )
         {
             sb.append( javadocPluginVersion ).append( ":" );
@@ -6370,6 +6364,8 @@ public abstract class AbstractJavadocMojo
                 getLog().info(
                     "The goal '" + javadocGoal + "' has not been previously called for the module: '" + p.getId()
                         + "'. Trying to invoke it..." );
+                
+                getLog().info( "External call using javadoc goal: " + javadocGoal );                
 
                 File invokerDir = new File( project.getBuild().getDirectory(), "invoker" );
                 invokerDir.mkdirs();
-- 
2.30.1 (Apple Git-130)

